File: 序号/429.N叉树的层序遍历.kt
class Solution {
  fun levelOrder(root: Node?): List<List<Int>> {
    // n叉树 层序遍历 bfs
    if (root == null) return emptyList()
    val ans = mutableListOf<MutableList<Int>>()
    var sum = root.children.size
    mutableListOf<Int>().toList()
    ans.add(mutableListOf(root.`val`))
    val q = ArrayDeque<Node>()
    q.addAll(root.children)
    while (q.isNotEmpty()) {
      val tmp = mutableListOf<Int>()
      var nextSum = 0
      for (i in 1..sum) {
        val node = q.poll()
        tmp.add(node.`val`)
        if (node.children.isNotEmpty()) {
          q.addAll(node.children)
          nextSum += node.children.size
        }
      }
      sum = nextSum
      ans.add(tmp)
    }
    return ans
  }
}

File: 序号/429. N 叉树的层序遍历.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        if not root:return []
        ans=[]
        q = deque([root])
        while q:
            l = []
            for _ in range(len(q)):
                node = q.popleft()
                l.append(node.val)
                if not node.children:continue
                for child in node.children:
                    q.append(child)
            ans.append(l)
        return ans